src/app/shared/models/workshop.model.ts

```diff
@@ -82,91 +78,113 @@ export class Workshop implements WorkshopTruncated {
     if (description.disabilityOptionsDesc) {
       this.disabilityOptionsDesc = description.disabilityOptionsDesc;
     }
-    if (description.imageFiles?.length) {
-      this.imageFiles = description.imageFiles;
-    }
-    if (description.imageIds?.length) {
-      this.imageIds = description.imageIds;
-    }
-    if (about.coverImage?.length) {
-      this.coverImage = about.coverImage;
-    }
-    if (about.coverImageId?.length) {
+  }
+}
+
+export class Workshop extends WorkshopBase {
+  takenSeats: number;
+  rating: number;
+  numberOfRatings: number;
+  status: WorkshopOpenStatus;
+  isBlocked: boolean;
+  providerOwnership: OwnershipTypes;
+  providerStatus: ProviderStatuses;
+
+  constructor(about: any, description: any, address: Address, teachers: Teacher[], provider: Provider, id?: string) {
+    super(about, description, address, teachers, provider, id);
+  }
+}
+
+export class WorkshopV2 extends Workshop {
```

Comment:
We need to think more about the name, v2 is not what we need, because it's actually the same version, but with additional content.
Is there any reason why we can't add this to the Workshop model itself?